rules/TypeDeclaration/NodeAnalyzer/ReturnTypeAnalyzer/StrictScalarReturnTypeAnalyzer.php

```diff
@@ -22,8 +25,11 @@ public function __construct(
     ) {
     }
 
-    public function matchAlwaysScalarReturnType(ClassMethod|Closure|Function_ $functionLike, Scope $scope): ?Type
-    {
+    public function matchAlwaysScalarReturnType(
```

Comment:
I'm not sure whether these classes / methods are part of a public API. In any case:
* `Scope $scope` already was unused, so could have been removed, but I kept it to not break any possible backwards compatibility
* `$hardCodedOnly = false` with default value to not break BC as well

So could be changed to this, if BC is not needed:
```suggestion
    public function matchAlwaysScalarReturnType(ClassMethod|Closure|Function_ $functionLike, bool $hardCodedOnly): ?Type
    {
```

Reply:
`Scope $scope` has been removed on master. Kept the `= false` default for `$hardCodedOnly`. Still not sure if needed though.

Reply:
add param is ok, just add default = false

Reply:
So as it currently is? 👍🏻

Reply:
Yes 👍